### DISCUSSION TITLE
Requesting updating outdated bottle

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5409] [Everyday usage]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I was trying to work with libimobiledevice on my system but it gave error when I tried to take backup. After research I  learned that bottle saved on homebrew is outdated contains a bug which latest code does not contain. When I tried using --head command and recompiled the library it worked perfectly. Now I needed it on another mac but I had to update my dev tools in order to build it again. Is there any way to request homebrew to update the bottles so they contain the latest code?

**Comment by SMillerDev** [comment]
> Is there any way to request homebrew to update the bottles so they contain the latest code?

https://docs.brew.sh/Updating-Software-in-Homebrew

**Reply by ghost** [reply]
Above link only mentions about formula and cask, but does not say anything about bottles.

**Comment by carlocab** [comment]
We ship the latest released version, which, at the time of writing is 1.3.0.[^1] This is the same version that the bottles are built for.

If you need a new release, please ask upstream to make a new one.

[^1]: https://libimobiledevice.org/#downloads

**Reply by ghost** [reply]
I have checked and bottle code behaves differently compared to one installed via --head command. I have opened issue with libimobiledevice maintianers as well but didn't get any response.

**Reply by p-linnane** [reply]
It behaves differently because there are over 200 commits that have been merged but not tagged in a release. I previously opened an issue almost a year ago asking for a new release: https://github.com/libimobiledevice/libimobiledevice/issues/1435

**Reply by ghost** [reply]
Is it possible the after I install it using brew install --head and then bottle the new build, saving it in custom tap. I tried doing following:
brew install --head --build-bottle libimobiledevice && brew bottle libimobiledevice

but after second command I got error that formula couldn't be found.
 
 It will be very helpful if you can leave a guide or point me to resource which achieve this. I tried reading official documentation but was unable to understand what I was doing wrong.

**Reply by SMillerDev** [reply]
I'm pretty sure head builds can't be bottled. You really just need to install with `--head` until upstream tags a release

**Reply by ghost** [reply]
Is there any other work around to avoid compiling every time I need to install the library? Maybe I can save the compiled code somewhere and write a script to replace current code with my required code.

**Reply by SMillerDev** [reply]
You could make a fork of the repo, tag that and bottle it in a tap.
